gdpr/anonymizers/base.py

```diff
@@ -0,0 +1,169 @@
+        raise NotImplementedError
+
+    def get_related_model(self) -> Model:
+        raise NotImplementedError
+
+
+class FieldAnonymizer:
+    """
+    Field anonymizer's purpose is to anonymize model field according to defined rule.
+    """
+
+    ignore_empty_values: bool = True
+    empty_values: List[Any] = [None]
+    _encryption_key = None
+    is_reversible: bool = True
+
+    class IrreversibleAnonymizationException(Exception):
+        pass
+
+    def __init__(self, ignore_empty_values: bool = None, empty_values: Optional[List[Any]] = None):
+        """
+        Args:
+            ignore_empty_values: defines if empty value of a model will be ignored or should be anonymized too
+            empty_values: define list of values which are considered as empty
+        """
+        self._ignore_empty_values = ignore_empty_values if ignore_empty_values is not None else self.ignore_empty_values
+        self._empty_values = empty_values if empty_values is not None else self.empty_values
+
+    def get_is_reversible(self, obj=None) -> bool:
+        """This method allows for custom implementation."""
+        return self.is_reversible
+
+    def get_anonymized_value_from_obj(self, obj, name: str, encryption_key: Optional[str] = None):
+        value = getattr(obj, name)
+        if self._ignore_empty_values and value in self._empty_values:
+            return value
+        self._encryption_key = encryption_key
+        return self.get_anonymized_value(value)
+
+    def get_deanonymized_value_from_obj(self, obj, name: str, encryption_key: Optional[str] = None):
+        if self.get_is_reversible(obj) is False:
+            raise self.IrreversibleAnonymizationException()
+        value = getattr(obj, name)
+        if self._ignore_empty_values and value in self._empty_values:
+            return value
+        self._encryption_key = encryption_key
+        return self.get_decrypted_value(value)
+
+    def get_anonymized_value_from_version(self, obj, version, name: str, encryption_key: Optional[str] = None):
+        value = version.field_dict[name]
+        if self._ignore_empty_values and value in self._empty_values:
+            return value
+        self._encryption_key = encryption_key
+        return self.get_anonymized_value(value)
+
+    def get_deanonymized_value_from_version(self, obj, version, name: str, encryption_key: Optional[str] = None):
+        if self.get_is_reversible(obj) is False:
+            raise self.IrreversibleAnonymizationException()
+        value = version.field_dict[name]
+        if self._ignore_empty_values and value in self._empty_values:
+            return value
+        self._encryption_key = encryption_key
+        return self.get_decrypted_value(value)
+
+    def get_encryption_key(self):
+        """Return encryption key to use.
+
+        Override this method for custom logic.
+        """
+        if self._encryption_key is None:
+            raise ImproperlyConfigured()
```

Comment:
tohle jen trošku ukazuje proč je to špatně, tohle by se nemělo dít. Můžeš nějak popsat proč to děláš takto složitě?